### ISSUE TITLE
Unable to apply containsstring on a property in an array

### ISSUE THREAD

**Issue by srane804** [WorkMaze/JUST.net#230]
I have an array like this :

**{
    "Id": "b2223b13-75d4-42c2-8fcf-183f2d8f9b2a",
    "FormNumber": "PSXS 01 06  07 20",
    "Notes": null
}, 
{
    "Id": "ee376a70-06ae-4272-a929-7405cd6f29fb",
    "FormNumber": "PSXS 01 12 07 20",
    "Notes": null
},
{
"Id": "b2223b13-75d4-42c2-8fcf-183f2d8f9b2r",
"FormNumber": "PCTC 01 06 07 20",
"Notes": null
},**

Now basically I want to filter out only those items where formnumber contains string "PSXS". So I have implemented the same as below : 
  **"FormsListCoverages": {
    "#loop($.Quotes[0].Coverages[?(#stringcontains(#valueof(@.FormNumber),PSXS),True)])": {
      "FormNumber": "#currentvalueatpath($.FormNumber)",
      "FormNotes": "#currentvalueatpath($.Notes)",
    }
  }**

I get an error as below when I try to run this : 
**Error: Unexpected character while parsing path query: #**

So basically I want to filter out those items in the Coverages array where FormNumber contains string 'PSXS' (i.e. first 2 records in above example).
It seems like the execution fails once it encounters the first # inside the coverages array.
Can you pls help me in making this work ?

**Comment by Courela** [comment]
This should do it:

```JSON
{
  "FormsListCoverages": {
    "#loop($.Quotes[0].Coverages[?(@.FormNumber =~ /^PSXS.*$//)])": {
      "FormNumber": "#currentvalueatpath($.FormNumber)",
      "FormNotes": "#currentvalueatpath($.Notes)",
    }
  }
}
```

On the #loop filter just use the regex operator (=~) to select the items you want.
Just one side note, since parenthesis is a reserved character, '/)' returns just the parenthesis, so one needs to escape slash character, that's why the regex expression is /^PSXS.*$//

**Comment by Courela** [comment]
First slash from regex expression is not considered an escape character since the following char ^ is not a reserved char. Thinking about it, every language I know does not take into account the next char when an escape char is found, if next char is not a recognised char to be escaped, the expression just fails... This requires the regex expression to be like //^PSXS.*$//

It will also work in this case, although the solution that I've given would fail.

**Comment by srane804** [comment]
Thanks @Courela for your reply. I tried the regex you suggested but it does not seem to work.

My filter is as below :
#loop($.Quotes[0].Coverages[?(@.FormNumber =~ //^PSXS.*$//)])

But I get this error while parsing : 
Error: Unexpected character while parsing path query: ^

**Comment by Courela** [comment]
Strange, both regexs worked for me, which version are you using?

**Comment by Courela** [comment]
That version should have already the escape char implemented... if it is working like that, the better, seems there's no need to escape slashes.

**Comment by srane804** [comment]
Thanks for the help @Courela. I will close this ticket
